Add onConfigure prop to table component

paulsturgess opened the pull request:
Adds an `onConfigure` prop to the table component. This displays a configure icon to the left of the table header and triggers the callback `onClick`.

Note I have changed the colour of the border that runs under the table header, otherwise there is a colour clash where the rounded border meets.

![screen shot 2017-06-13 at 08 56 11](https://user-images.githubusercontent.com/4964/27071796-354c29f6-5016-11e7-956e-65db993acb2d.png)

![screen shot 2017-06-13 at 08 44 23](https://user-images.githubusercontent.com/4964/27071798-3753a13e-5016-11e7-8985-890e06beca87.png)

#### TODO
- [ ] Changelog
- [x] Specs

toni-leigh commented:
@paulsturgess It's good to put detail like that in the PR description into the commit message itself, then it goes with the repo and will show up in the history tools, very useful if someone is digging to find out what and why

If you create a PR with just one commit, then github even gets the commit details and puts them in the description

toni-leigh commented on src/components/table/table.scss:
```diff
@@ -39,3 +48,30 @@
+    border-right: none;
+    border-radius: 25px 0 0 25px;
+    color: $white;
+    display: flex;
```
This is working with `margin: auto;` but I can pre-empt a weird IE11 + WIN10 bug - use:

```
align-items: center;
justify-content: center;
```

in the same selector as `display: flex;` and remove the margin definition from the child.

toni-leigh commented on src/components/table/table.scss:
```diff
@@ -39,3 +48,30 @@
+      color: $white;
+    }
+
+    span {
```
I expect this can be fully removed given the above comment, but for future reference using the `span` created by a component is risky as it might change, a `className` on the Carbon component (LInk in this case?) would be better

paulsturgess commented:
@toni-leigh I've updated the css and commit message as requested. I still need to target the span wrapper in the icon to shift it a bit as the center alignment positions it too close to the rounded left edge.

As the icon span wrapper doesn't have a css class attached, i've switched to using `:first-child`. Hopefully this should be more robust. If this is not reasonable then I think we'll need to update the icon component (unless you can see another way).

I looked at updating the alignment of the text in the header and you are right it is not vertically in the center, however, I think it's positioned like this on purpose. I adjusted the positioning (setting the top and bottom padding to be equal) it looked really odd. So I've left this as-is. Unfortunately I think the cog icon height doesn't quite match perfectly the font-size in the header, so it's always 1px slightly too high/low to be absolutely perfect.

toni-leigh commented:
This marked up image is take from compendium which is what this work is copying

![image](https://user-images.githubusercontent.com/10499070/27037715-ad7a4624-4f80-11e7-9219-6c9398625db4.png)

Maybe try centring with flex to avoid fixed pixel padding changes, that might do some clever thing with pixels to exactly centre text?

paulsturgess commented:
@toni-leigh I found that by removing the `line-height` and the top and bottom padding from the table header styles the text is now aligning itself vertically in the center. I've updated the screengrabs in the description.

toni-leigh commented:
@paulsturgess nice

Line height and pixel paddings are both hacks for vertical centring, good to remove them,

Flex should be the go-to tool for vertical centring, I'm guessing flex isn't being used here though?

paulsturgess commented:
@toni-leigh was there anything else you wanted me to look at on this PR?

abramin commented on src/components/table/__spec__.js:
```diff
@@ -685,6 +685,35 @@ describe('Table', () => {
   });
 
+  describe('onConfigure', () => {
+    let onConfigureSpy = jasmine.createSpy();
```
We're trying to write all new tests in Enzyme

paulsturgess replied:
I've re-written the new specs to use Enzyme

SeanArmstrong commented:
👍
